build/utils.go

```diff
@@ -67,11 +67,16 @@ func toBuildkitExtraHosts(inp []string, mobyDriver bool) (string, error) {
 		if !ok || host == "" || ip == "" {
 			return "", errors.Errorf("invalid host %s", h)
 		}
-		// Skip IP address validation for "host-gateway" string with moby driver
-		if !mobyDriver || ip != mobyHostGatewayName {
-			if net.ParseIP(ip) == nil {
-				return "", errors.Errorf("invalid host %s", h)
+		// If the IP Address is a "host-gateway", replace this value with the
+		// IP address provided by the worker's label.
+		if ip == mobyHostGatewayName {
+			if hgip == nil {
```

Comment:
Would it make  sense to just pass a reference to the `nodeDriver` as argument, and do the lookup here? (Could even be more generic and a `nodeDriver.LookupIP(name)` perhaps, but maybe that's not needed). And we could return the error it returned here. (Function could return a `NotFound()` error if not supported, but not sure about that one)